aws/table_aws_cloudtrail_lake_query.go

```diff
@@ -0,0 +1,199 @@
+package aws
+
+import (
+	"context"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudtrail"
+	"github.com/aws/aws-sdk-go-v2/service/cloudtrail/types"
+
+	"github.com/turbot/steampipe-plugin-sdk/v4/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsCloudtrailLakeQuery(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_cloudtrail_lake_query",
+		Description: "AWS CloudTrail Lake Query",
+		List: &plugin.ListConfig{
+			ParentHydrate: listCloudTrailEventDataStores,
+			Hydrate:       listCloudtrailLakeQueries,
+			KeyColumns: plugin.KeyColumnSlice{
+				{
+					Name:    "event_data_store_arn",
+					Require: plugin.Optional,
+				},
+			},
+		},
+		GetMatrixItemFunc: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "query_id",
```

Comment:
We should support `query_id` as an optional key column

Reply:
The list call does not support `QueryId` as an input param

Reply:
@cbruno10 The `DescribeQuery` API doesn't provide all the properties we are getting from list call, so we can not use it in `GetConfig`. We can use this API as a hydrate call to fetch some other information that is expressly provided by this API.